flang/lib/Semantics/resolve-names.cpp

```diff
@@ -3797,6 +3797,17 @@ bool SubprogramVisitor::Pre(const parser::PrefixSpec::Attributes &attrs) {
         subp->set_cudaSubprogramAttrs(attr);
       }
     }
+    if (auto attrs{subp->cudaSubprogramAttrs()}) {
+      if (*attrs == common::CUDASubprogramAttrs::Global ||
+          *attrs == common::CUDASubprogramAttrs::Device) {
+        // Implicitly USE the cudadevice module by copying its symbol in the
+        // current scope.
```

Comment:
s/symbol/symbols/

What about clashes with names that are already in scope (or declared later)?

Reply:
So according to the reference compiler we should not overwrite what is already in scope so I updated the copy of symbol and added a test.
For what is declared after, in device or global procedure, we will have an error msg. I added a test also for this case.

Reply:
So the semantics of these names are not really what you get from either a USE statement or from intrinsics.

There is already precedence here with names like `threadIdx` that are automatically imported into device subprogram scopes, so I guess this isn't making things any worse.